python_modules/dagster/dagster/_core/execution/context/system.py

```diff
@@ -342,59 +342,96 @@ def scoped_resources_builder(self) -> ScopedResourcesBuilder:
     def log(self) -> DagsterLogManager:
         return self._log_manager
 
+    @property
+    def partitions_def(self) -> Optional[PartitionsDefinition]:
+        from dagster._core.definitions.job_definition import JobDefinition
+
+        pipeline_def = self._execution_data.pipeline_def
+        if not isinstance(pipeline_def, JobDefinition):
+            check.failed(
+                "Can only call 'partitions_def', when using jobs, not legacy pipelines",
+            )
+        partitions_def = pipeline_def.partitions_def
+        return partitions_def
+
+    @property
+    def is_partitioned_run(self) -> bool:
```

Comment:
what do you think about calling this `has_partitions` for consistency with `has_asset_partitions`?

Reply:
sounds good! changed this